### PR TITLE
Add different logging format

### PR DESCRIPTION
**Pull request by GuanLuo**
Default
```
I0523 23:49:06.674147 141 server.cc:288] All models are stopped, unloading models
```
ISO8601
```
2022-05-23T23:49:14 I 153 server.cc:288] All models are stopped, unloading models
```

### CODE REVIEW

**Review by dyastremsky — APPROVED** [review]

**Review by krishung5 — APPROVED** [review]

File: include/triton/common/logging.h [hunk 1]
```diff
@@ -51,6 +51,8 @@ class LogMessage {
 // Global logger for messages. Controls how log messages are reported.
 class Logger {
  public:
+  enum Format { kDEFAULT = 0, kISO8601 = 1 };
```
Comment (rmccorm4) [inline include/triton/common/logging.h:54]: `enum class` instead? https://stackoverflow.com/a/18335862

**Comment by tanmayv25** [comment]
Not required, but should we add a time zone designator in the timestamp? Looks like ISO 8601 goes about at length regarding the same. I would suggest `Local time with an offset to UTC`. 
It will be helpful for people to correlate logs from across the timezones.

File: src/logging.cc [hunk 2]
```diff
@@ -70,30 +70,67 @@ LogMessage::LogMessage(const char* file, int line, uint32_t level)
...
+            << std::setw(2) << tm_time.tm_mday << " " << std::setw(2)
+            << tm_time.tm_hour << ':' << std::setw(2) << tm_time.tm_min << ':'
+            << std::setw(2) << tm_time.tm_sec << "." << std::setw(6) << tv.tv_usec
+            << ' ' << static_cast<uint32_t>(getpid()) << ' ' << path << ':'
+            << line << "] ";
 #endif
+    break;
+  }
+  case Logger::Format::kISO8601: {
+    // YYYY-MM-DDThh:mm:ss L
```
Comment (rmccorm4) [inline src/logging.cc:105]: Maybe an addition of a `Z` specifier at the end to denote this is in UTC would be helpful, similar to Tanmay's comment.

> As format Z,  This date time has no offset from UTC (or 0 hour offset).

File: src/logging.cc [hunk 3]
```diff
@@ -70,30 +70,67 @@ LogMessage::LogMessage(const char* file, int line, uint32_t level)
     path = path.substr(pos + 1, std::string::npos);
   }
 
+  // 'L' for log level
```
[the remark this answers is not in the document]
Comment (rmccorm4) [reply src/logging.cc:73]: I believe it's saying the 'L' references in the following formats means Level:
i.e.
```
  case Logger::Format::kDEFAULT: {
    // LMMDD hh:mm:ss.ssssss     <----
...
  }
  case Logger::Format::kISO8601: {
    // YYYY-MM-DDThh:mm:ss L     <----
...
```

File: src/logging.cc (hunk 2 above)
Comment (GuanLuo) [reply src/logging.cc:105]: @tanmayv25 I think I will use `Z` to highlight UTC time as described here. The doc for `gettimeofday` suggests that there can be complication when handling local time.
> The use of the timezone structure is obsolete; the tz argument
       should normally be specified as NULL.  (See NOTES below.)

File: src/logging.cc (hunk 3 above)
[the remark this answers is not in the document]
Comment (dyastremsky) [reply src/logging.cc:73]: I see, thanks for clarifying!

**Review by rmccorm4 — APPROVED** [review]

**Review by krishung5 — APPROVED** [review]

**Review by tanmayv25 — APPROVED** [review]

**Review by tanmayv25 — APPROVED** [review]

**Review by dyastremsky — APPROVED** [review]

**Pull request merged**